src/main/java/graphql/util/Traverser.java

```diff
@@ -171,7 +124,8 @@ private Traverser(TraverserState<T> traverserState, Function<? super T, ? extend
                 }
             }
         }
-        return currentContext.getResult();
+        TraverserResult traverserResult = new TraverserResult(cycle, currentContext.getResult(), fullTraversal);
```

Comment:
in case the loop was broken due to `visitor.leave` method requested to `QUIT`, the currentContext will contain the result of the previous `visitor.backRef` or `visitor.enter` calls. Is that intentional? and why?

Reply:
additionally, in case `visitor.backRef` decides to continue the traversal, `cycle` variable will be set to true and when the traversal loop is ended either normally or normally, but not because of another cyclic reference, the last value of `cycle=true` will be reported as the result of the traversal. Is that intentional? and why?

Reply:
regarding the first point: good catch, that was not intended, that needs to be fixed and tested. thanks

about the second point: not sure I understand it correctly, but cycle true means we encountered an cycle regardless of how we deal with it. I guess cycle is not a good name here: in the `TraverserResult` it is called enounteredCycle which makes the intention a bit more clear I hope. 
And fullTraversal means we aborted or quit the tree. Does this makes sense?

Reply:
I tink I understand the intent, please, correct me if I am wrong. You want to report with the result if a traversal has encountered a cycle or not. 
I am thinking about the use case. What the client is expected to do with it?
If, let's say, the client needs to simply report that a cycle was encountered, then, perhaps, it needs to know what node has cyclic reference(s). 
Additionally, if there are many cycles detected, then perhaps, all of them need to be reported.

wdyt?

Reply:
I was thinking of some validation, where you wanna handle cycles explicitly.
But you are right: maybe I wanna all of them etc.
Because it is not used atm I will remove the cycle and the  `fullTraversal ` property again. Keep it simple for the moment. The important aspect was to introduce a proper Result class `TraverserResult`.